client/components/AllProducts.js

```diff
@@ -1,32 +1,41 @@
-import React from 'react';
+import React, {Component} from 'react';
 import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
 import Banner from './Banner'
+import Search from './Search'
 
-export const AllProducts = ({allProducts}) => {
-  return (
-    <div>
-    <Banner />
-      {
-        allProducts &&
-          allProducts.map(product => {
-            return (
-              <div key={product.id}>
-                <Link to={`/products/${product.id}`}>
-                  <img src={product.imageUrl} />
-                  <h2>{product.title}</h2>
-                </Link>
-              </div>
-            )
-          })
-      }
-    </div>
-  )
+export class AllProducts extends Component {
+  render() {
+    const { allProducts, searchInput } = this.props
+    const filteredProducts = searchInput.length ? allProducts.filter(product => product.title.toLowerCase().match(searchInput.toLowerCase())) : allProducts
```

Comment:
does this handle when we are not searching for anything? so for example if we don't have `searchInput` if it is blank then nothing will render since we are mapping over `filteredProducts` and if there is no `searchInput` there won't be anything to match it against

Reply:
The ternary checks for that logic, so if there is no searchInput it will default to allProducts